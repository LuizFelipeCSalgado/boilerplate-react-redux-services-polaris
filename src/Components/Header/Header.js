import React from 'react';
import PropTypes from 'prop-types';

import PageHeader from 'Components/PageHeader';

import './Header.scss';

function Header({ children }) {
  return (
    <header>
      <PageHeader />
      {children}
    </header>
  );
}

Header.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
};

Header.defaultProps = {
  children: null,
};

export default Header;
