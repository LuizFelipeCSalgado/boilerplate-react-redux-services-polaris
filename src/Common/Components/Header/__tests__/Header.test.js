import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Header from '../Header';

describe('Header', () => {
  let wrapper = null;
  const props = {};

  const mountComponent = (newProps = {}) => {
    wrapper = mount(<Header {...props} {...newProps} />);
  };

  beforeEach(() => {
    mountComponent();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Header {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

})
