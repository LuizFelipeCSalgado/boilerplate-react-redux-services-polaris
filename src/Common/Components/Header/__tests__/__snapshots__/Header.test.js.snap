// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the without crashing 1`] = `null`;
